lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -244,7 +244,7 @@ def formatted_message_and_backtrace(colorizer)
         end
 
         def exception_backtrace
-          exception.backtrace || []
+          RSpec.configuration.backtrace_formatter.format_backtrace(exception.backtrace) || []
```

Comment:
> Is there any way to include specific gem to project_source_dirs ?

`project_source_dirs` is a public API. Just update it in your `RSpec.configure` block.  For example:


``` ruby
RSpec.configure do |rspec|
  require 'my_cool_gem'
  rspec.project_source_dirs << $LOAD_PATH.grep(/my_cool_gem/).first
end
```

Of course, if you know exactly where your gem is located, you could simplify this, but querying `$LOAD_PATH` after loading the gem seems like a robust way to add the gem's `lib` directory to `project_source_dirs`.